src/TLuaInterpreter.cpp

```diff
@@ -9397,6 +9397,17 @@ int TLuaInterpreter::getRoomEnv( lua_State * L )
     return 0;
 }
 
+// Past code returned an empty string if room or user data item with given key
+// didn't exist - to enable the more modern behaviour that produces a
+// nil + error message (recommended) in those cases a third (boolean)true is
+// required. This "correct" behaviour being non-default is to retain backwards
+// compatibility with existing scripts/packages that expect an empty string,
+// even though the past code would allow the user to assign such an empty string
+// against a key which cannot be distinguished in such circumstances.
+// This fix is specific to the Room User Data as the need for it was introduced
+// at the same time as Area and Map User Data was added but they were not
+// documented until later and their Wiki entries did/will not mention returning
+// an empty string in the cases of no room with id or no key with given name.
```

Comment:
Well actually it would be "no area with id" and only apply to the area case anyhow but that is not really of great concern here...